test/integration/scheduler_perf/scheduler_perf.go

```diff
@@ -469,6 +469,14 @@ type createPodsOp struct {
 	// Number of pods to be deleted per second after they were scheduled. If set to 0, pods are not deleted.
 	// Optional
 	DeletePodsPerSecond int
+	// Scheduling throughput threshold is compared to average pods/s SchedulingThroughput metric.
```

Comment:
I bet I'll want to implement scheduler_perf to measure the event handling performance after https://github.com/kubernetes/kubernetes/pull/125929.
Also, some custom plugin developers use scheduler_perf outside k/k, maybe with their custom metrics.

So, from those perspective, I have several points -
1. Can we move this threshold setting to `testCase`?
2. Can we make it enable to change the target metric from  SchedulingThroughput?
3. Do we currently collect metrics (except `SchedulingThroughput`) during operations apart from CreatePodOp? 
If we don't, we should. But, this can be follow-up in another PR.

Reply:
Right, we can make it more generic:
1. I would keep the setting in specific op, especially if we would like to measure metrics for multiple ops in one workload, it would make it easier to differentiate the thresholds. I think, the values can also differ depending on the workload size, e.g. smaller one can have higher throughput than bigger. By using `throughputThresholdParam` per workload and then passing it into specific op, the thresholding can be more customizable.
2. It should be possible, but we need to agree how to describe the target metric in the config. There are `Metric`, `extension_point`, `plugin` and `result` fields that can differ depending on metric and plugin. I think we should have one field that can describe all four uniquely.
3. It is only possible to collect metrics during `createPods` op at the moment, but for sure it could be added in one of the next PRs.

Reply:
2. Do we want to support setting multiple thresholds for different metrics per one test case?

Reply:
I only express my opinion. I use scheduler_perf in my custom plugins to measure throughput.

Reply:
>  especially if we would like to measure metrics for multiple ops in one workload, it would make it easier to differentiate the thresholds.

Then, how about moving it to `workload`?

My point is that I don't want to put it into `createPodOp` as we'll likely have to support it in many other kinds of op. Also, I wouldn't like to give an assumption that having multiple `CollectMetrics:true` with `ThroughputThreshold` in one test case is allowed.

> It should be possible, but we need to agree how to describe the target metric in the config. By default, there are Metric, extension_point, plugin and result labels that can differ depending on metric and plugin. I think we should have one field that can describe them uniquely in the config. I think the map could be enough.

Yeah, we should have metric selector-ish something + threshold, something like:

```yaml
threshold: 1
metricSelector:
  name: scheduler_plugin_execution_duration_seconds
  labels: # map[string]string
    plugin: nodeaffinity
    extension_point: Filter
```

This can be follow-up; I'm OK to only support SchedulingThreshold in this PR, as a starting point.

Reply:
> Do we want to support setting multiple thresholds for different metrics per one test case?

This is a good point. 
I'd say, we might? But, IMO, no need to support multiple right now, at least for now we can just target one metric in one test case.

Reply:
> Also, I wouldn't like to give an assumption that having multiple CollectMetrics:true with ThroughputThreshold in one test case is allowed.

Follow-up description for this point.

Technically, we can specify multiple pod op with `CollectMetrics:true`, but, I bet currently we assume we have `CollectMetrics:true` only once during a test case; at least, I believe perf-dash couldn't handle such scenarios.

Also, I've gone through an implementation a bit; looks like native scheduler metrics (i.e., metrics except SchedulingThroughput) are always collected (i.e., during other ops without CollectMetrics:true too), and we just summarize data at createPods. (doesn't it?)
https://github.com/kubernetes/kubernetes/blob/master/test/integration/scheduler_perf/util.go#L266-L268

So, multiple `CollectMetrics:true` ops in a single test case don't work expectedly in all metrics except SchedulingThroughput; Let's say op1 and op2 have CollectMetrics:true, the op2's summary of metric values contain the values that have been collected during op1 (+ all the previous ops) too. It's not that op1's `CollectMetrics` only collects metric values during op1, and op2's one only during op2.

Reply:
Okay, to sum up:

- I'll make the threshold defined per workload, not specific op and compare it wherever the collectMetrics will be true.

- It will be able to select metric using `throughputMetricSelector` field as shown above. I think we could allow to set some default value for this field (e.g. `defaultThresholdMetricSelector`) at testCase level to omit redundancy. 

And:

> So, multiple `CollectMetrics:true` ops in a single test case don't work expectedly in all metrics except SchedulingThroughput; Let's say op1 and op2 have CollectMetrics:true, the op2's summary of metric values contain the values that have been collected during op1 (+ all the previous ops) too. It's not that op1's `CollectMetrics` only collects metric values during op1, and op2's one only during op2.

Do you think we should try to fix it in one of follow-ups?

Reply:
Done